lib/libc/gen/getpagesize.c

```diff
@@ -29,45 +29,24 @@
  * SUCH DAMAGE.
  */
 
-#include <sys/cdefs.h>
-__SCCSID("@(#)getpagesize.c	8.1 (Berkeley) 6/4/93");
 #include <sys/param.h>
-#include <sys/sysctl.h>
-
-#include <errno.h>
-#include <link.h>
-#include <unistd.h>
+#include <sys/auxv.h>
 
 #include "libc_private.h"
 
 /*
- * This is unlikely to change over the running time of any
- * program, so we cache the result to save some syscalls.
- *
  * NB: This function may be called from malloc(3) at initialization
  * NB: so must not result in a malloc(3) related call!
  */
 
 int
 getpagesize(void)
 {
-	int mib[2];
-	static int value;
-	size_t size;
-	int error;
-
-	if (value != 0)
-		return (value);
+	int error, value;
```

Comment:
init value to zero now that it is not in bss?

Reply:
Not necessary I think due to the `error != 0` fallback (which in practice is dead code)?

Reply:
Oh yes I'm not sure how I misread it.